subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -391,6 +391,17 @@ static struct bass_recv_state_internal *bass_lookup_addr(const bt_addr_le_t *add
 	return NULL;
 }
 
+static struct bass_recv_state_internal *bass_lookup_broadcast_id(const uint32_t broadcast_id)
+{
+	for (int i = 0; i < ARRAY_SIZE(scan_delegator.recv_states); i++) {
```

Comment:
needs to guard scan_delegator.recv_states to avoid the BASS overwritting

Reply:
As discussed, this is a general issue with the API, and Ill leave this for future fixing